File: irrgartenjaava/src/irrgartenjaava/FuzzyPlayer.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package irrgartenjaava;

import java.util.ArrayList;

/**
 *
 * @author jxlig0d
 */
public class FuzzyPlayer extends Player{
    
    public FuzzyPlayer (Player other){
        super(other);
    }

    @Override
    public Directions move(Directions direction, ArrayList<Directions> validMoves){
        Directions preference=super.move(direction, validMoves);
        return Dice.nextStep(preference, validMoves, this.getIntelligence());
    }
    
    @Override
    public float attack(){
        return (Dice.intensity(this.getStrength())+this.sumWeapons());
    }
    
    @Override
    protected float defensiveEnergy(){
        return (Dice.intensity(this.getIntelligence())+this.sumShields());
    }
    
    @Override
    public String toString(){
        String toReturn="((Fuzzy)) ";
        toReturn+=super.toString();
        
        return toReturn;
    }
}

File: irrgarten_ruby/src/fuzzy_player.rb
# encoding: UTF-8

require_relative 'player'
require_relative 'dice'


module Irrgarten

    class FuzzyPlayer < Player

        # Constructor de la clase {FuzzyPlayer}.
        def initialize(other_player)
            copy(other_player)
        end

        def move(direction, valid_moves)

          Dice.next_step(super, valid_moves, intelligence)
       end

        def attack
            return self.sum_weapons + Dice.intensity(@strength)
        end

        def to_s
            return "(Fuzzy) " + super
        end

        protected
        
        def defensive_energy
            return self.sum_shields + Dice.intensity(@intelligence)
        end

    end # class
end # module